test/trento/application/process_managers/deregistration_process_manager_test.exs

```diff
@@ -2,24 +2,55 @@ defmodule Trento.DeregistrationProcessManagerTest do
   use ExUnit.Case
 
   alias Trento.Domain.Events.{
+    ApplicationInstanceDeregistered,
+    ApplicationInstanceRegistered,
     ClusterRolledUp,
+    DatabaseInstanceDeregistered,
+    DatabaseInstanceRegistered,
     HostAddedToCluster,
     HostDeregistered,
     HostDeregistrationRequested,
     HostRegistered,
     HostRemovedFromCluster,
-    HostRolledUp
+    HostRolledUp,
+    SapSystemRolledUp
   }
 
   alias Trento.DeregistrationProcessManager
-  alias Trento.Domain.Cluster
+
+  alias Trento.DeregistrationProcessManager.Instance
+
+  alias Trento.Domain.{
+    Cluster,
+    SapSystem
+  }
 
   alias Trento.Domain.Commands.{
+    DeregisterApplicationInstance,
     DeregisterClusterHost,
+    DeregisterDatabaseInstance,
     DeregisterHost
   }
 
   describe "events interested" do
+    test "should start the process manager when DatabaseInstanceRegistered event arrives" do
```

Comment:
Let's try to put a correct order. This tests are last to add, so put them at the botton of this `describe` please